kimchi/src/circuits/polynomials/poseidon.rs

```diff
@@ -76,30 +87,28 @@ pub fn constraint<F: FftField + SquareRootField>() -> E<F> {
         })
         .collect();
 
-    for e in ROUND_EQUATIONS.iter() {
-        let &RoundEquation {
-            source,
-            target: (target_row, target_round),
-        } = e;
-        let sboxed: Vec<_> = round_to_cols(source)
-            .map(|i| {
-                cache.cache(
-                    E::cell(Column::Witness(i), Curr).pow(PlonkSpongeConstants15W::SPONGE_BOX),
-                )
-            })
-            .collect();
+    for (round, eq) in ROUND_EQUATIONS.iter().enumerate() {
+        // sbox
+        let mut sboxed = vec![];
```

Comment:
The size of this vector is known in advance, use `with_capacity` to avoid allocations with `push`.